File: src/shape_embedding_training/mirror_shape_embedding_training.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import os
import sys

BASE_DIR = os.path.dirname(os.path.abspath(__file__))
sys.path.append(os.path.dirname(BASE_DIR))
from global_variables import *

shapenet_synset_set_handle_mirror = g_shapenet_synset_set_handle + '_' + g_mirror_name;
shape_list_mirror_mapping_filename = os.path.join(g_data_folder, 'shape_list_mapping'+shapenet_synset_set_handle_mirror+'.txt')
print 'Loading shape list mapping from %s...'%(shape_list_mirror_mapping_filename)
shape_list_mirror_mapping = [bool(int(line.strip())) for line in open(shape_list_mirror_mapping_filename, 'r')]

print 'Loading original shape embedding space from %s...'%(g_shape_embedding_space_file_txt)
embedding_space = [line.strip() for line in open(g_shape_embedding_space_file_txt, 'r')]

shape_embedding_space_file_txt_mirror = os.path.join(g_data_folder, 'shape_embedding/shape_embedding_space'+shapenet_synset_set_handle_mirror+'.txt')
print 'Saving mirror shape embedding space to %s...'%(shape_embedding_space_file_txt_mirror)
with open(shape_embedding_space_file_txt_mirror, 'w') as shape_embedding_space_file_txt_mirror_file:
    for idx, mirror in enumerate(shape_list_mirror_mapping):
        if mirror:
            shape_embedding_space_file_txt_mirror_file.write(embedding_space[idx]+'\n')

File: src/shape_embedding_training/mirror_shape_embedding_training.m
addpath('../');
global_variables;

%% Load shape distance matrix
t_begin = clock;
fprintf('Loading shape distance matrix from \"%s\"...', g_shape_distance_matrix_file_mat);
load(g_shape_distance_matrix_file_mat);
t_end = clock;
fprintf('done (%f seconds)!\n', etime(t_end, t_begin));

apenet_synset_set_handle_mirror = [g_shapenet_synset_set_handle '_' g_mirror_name];
shape_embedding_space_file_mat_mirror = fullfile(g_data_folder, ['shape_embedding/shape_embedding_space' apenet_synset_set_handle_mirror '.mat'])
shape_embedding_space_file_txt_mirror = fullfile(g_data_folder, ['shape_embedding/shape_embedding_space' apenet_synset_set_handle_mirror '.txt'])

shape_list_mirror_mapping_filename = fullfile(g_data_folder, ['shape_list_mapping' apenet_synset_set_handle_mirror '.txt'])
shape_list_mirror_mapping = importdata(shape_list_mirror_mapping_filename);

mirror_mapping = logical(shape_list_mirror_mapping*shape_list_mirror_mapping');
shape_distance_matrix = squareform(shape_distance_matrix);
shape_distance_matrix = shape_distance_matrix(mirror_mapping);
shape_distance_matrix = squareform(shape_distance_matrix);


%% Save shape distance matrix
t_begin = clock;
fprintf('Save shape distance matrix to \"%s\"...', shape_embedding_space_file_mat_mirror);
save(shape_embedding_space_file_mat_mirror, 'shape_distance_matrix', '-v7.3');
shape_distance_matrix_NxN = squareform(shape_distance_matrix);
dlmwrite(shape_embedding_space_file_txt_mirror, shape_distance_matrix_NxN, 'delimiter', ' ');
t_end = clock;
fprintf('done (%f seconds)!\n', etime(t_end, t_begin));

exit;